src/expr/src/vector_op/extract.rs

```diff
@@ -12,35 +12,45 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-use chrono::{Datelike, Timelike};
+use chrono::{Datelike, NaiveDate, NaiveTime, Timelike};
 use risingwave_common::types::{Date, Decimal, Time, Timestamp};
 use risingwave_expr_macro::function;
 
 use crate::{ExprError, Result};
 
-fn extract_time<T>(time: T, unit: &str) -> Option<Decimal>
-where
-    T: Timelike,
-{
+fn extract_time(time: NaiveTime, unit: &str) -> Option<Decimal> {
     Some(match unit {
-        "HOUR" => time.hour().into(),
-        "MINUTE" => time.minute().into(),
-        "SECOND" => time.second().into(),
+        "HOUR" | "hour" => time.hour().into(),
+        "MINUTE" | "minute" => time.minute().into(),
+        "SECOND" | "second" => time.second().into(),
```

Comment:
The old implementation is actually wrong. `second` returns fractional part in PostgreSQL.